packages/clay-layout/src/Container.tsx

```diff
@@ -0,0 +1,61 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import React from 'react';
+
+interface IProps extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Adds `.container-fluid` class to create a fluid container that
+	 * doesn't expand beyond a set width
+	 */
+	fluid?: boolean;
+
+	/**
+	 * Adds the `.container-form-${formSize}` class to properly space
+	 * between application controls and the form. This class only modifies
+	 * the padding on the container.
+	 */
+	formSize?: 'sm' | 'md' | 'lg' | 'xl';
+
+	/**
+	 * Adds `.container-fluid-${size}` class to set max width on container.
+	 */
+	fluidSize?: 'sm' | 'md' | 'lg' | 'xl';
+
+	/**
+	 * Adds the `.container-view` class to properly space between application
+	 * controls and view pages (e.g., Card View, Table View, or List View).
+	 * This class only modifies the padding on the container.
+	 */
+	view?: boolean;
```

Comment:
Same as with `.formSize`...

@eduardoallegrini, thoughts?

Reply:
I'm unsure as to the importance of these "proprietary" classes if we're going with the approach of separating the library from Portal. If I were making an app using Clay, I wouldn't really bother too much with props that have these very specific use cases.

Perhaps the question to ask is: _"How often will the user use this component with pre-existing knowledge of which props they'll need?"_
If it were me, I'd use `<ClayContainer flexOrEquivalent />` and then build out the rest, and then when it came to styling I'd probably not go back and check every component to see if I can configure the layout using props but instead just work in CSS.

Reply:
> I were making an app using Clay, I wouldn't really bother too much with props that have these very specific use cases [...]
> 
> [...] I'd probably not go back and check every component to see if I can configure the layout using props but instead just work in CSS [...]

That begs the question... why would you bring in a library dependency such as Clay to substitute `<div>` by `<ClayContainer>` and keep it all the same?

If you won't bother checking the props and defaults of the library of your choosing, why'd you choose it in the first place? 

We've discussed at length our vetting process for picking dependencies that are actually worth their value. If we were to have that conversation around this library, maybe @wincent, or yourself would argue that we don't need Clay since we already have `flexOrEquivalent`.

That might mean, you wouldn't be a user of Clay in a normal situation, so we shouldn't model it around this preference :)

Reply:
The fact that the primary goal of Clay is to serve DXP makes my concern kinda obsolete. And that's fine, I was just looking out for the little man here 😄, all of us using Clay while building DXP will look into all the props and use them accordingly. So having the classes that are omnipresent throughout DXP  in this component as props is a time saver for us.

Reply:
> primary goal of Clay is to serve DXP

I didn't say this!!

**Primary goal of Clay is to make building Lexicon UIs easy, consistent, robust and safe.**

DXP is the main consumer right now, so we shouldn't fall into the common trap of serving imaginary users and failing to deliver to those teams. We've been down that path before. It ain't pretty 😉 

That being said, we also aim for other departments and teams to jump on board as Design practices converge. AC (Faro), IS, DXP Cloud... they will all benefit in the long run if we do a good job here. That's why looking out for those "general" users is still valid and important, just not at the expense of DXP.

Reply:
Yea, I was worried my wording was too strong. I get what you're saying 👍 . I was focusing on the imaginary users more than on DXP. That's what makes Lexicon great, as long as we make components based on their design it should be great for both DXP and the other projects Liferay has, right?

Reply:
> That's what makes Lexicon great, as long as we make components based on their design it should be great for both DXP and the other projects Liferay has, right?

That's the dream 😂

Reply:
Similar to `formSize`... I know this is being used, si this has indeed real backing, but I wouldn't commit to it too early.